### PR TITLE
add checkbutton function

### PR DESCRIPTION
**Pull request by Korbin-Wang**
Please

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import MDButtonGroup from './src/MDButtonGroup.jsx';
+import MDButtonGroup from './src/MDButtonGroup.js';
```
Comment (jimexist) [inline App.js]: why?

File: lib/MDButtonGroup.css [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* put your css here */
+:local .container .clicked {
+    color: pink;
```
Comment (jimexist) [inline lib/MDButtonGroup.css:3]: why pink...

File: package.json [hunk 3]
```diff
@@ -34,6 +34,7 @@
     "babel-preset-react": "6.11.1",
     "babel-preset-react-hmre": "1.1.1",
     "babel-preset-stage-0": "6.5.0",
+    "classnames": "^2.2.5",
```
Comment (jimexist) [inline package.json:37]: also seems to me a peer dependency? no?

File: src/MDButtonGroup.js [hunk 4]
```diff
@@ -10,11 +11,12 @@ class MDButtonGroup extends Component {
       activeButtons: new Set([]),
     }
     this.onRadioClick = this.onRadioClick.bind(this);
+    this.onCheckClick = this.onCheckClick.bind(this);
   }
 
   onRadioClick(e) {
     e.preventDefault();
-    const { activeButtons } = this.state;
+    const { activeButtons } = this.state;   //destruction
```
Comment (jimexist) [inline src/MDButtonGroup.js]: you don't need these comments

File: src/MDButtonGroup.js [hunk 5]
```diff
@@ -26,12 +28,24 @@ class MDButtonGroup extends Component {
     });
   }
 
+  onCheckClick(e) {
+    e.preventDefault();
+    const { activeButtons } = this.state;
+    const { innerHTML} = e.target; // e.target： object
```
Comment (jimexist) [inline src/MDButtonGroup.js]: you don't need these comments

Comment (jimexist) [inline src/MDButtonGroup.js]: also style: space before `}`

File: src/MDButtonGroup.js [hunk 6]
```diff
@@ -26,12 +28,24 @@ class MDButtonGroup extends Component {
...
+    if (activeButtons.has(innerHTML)){
+      activeButtons.delete(innerHTML);
+    }else{
```
Comment (jimexist) [inline src/MDButtonGroup.js]: style: spaces

File: src/MDButtonGroup.js [hunk 7]
```diff
@@ -26,12 +28,24 @@ class MDButtonGroup extends Component {
...
+    this.forceUpdate();
+  }
+
   getButtonGroup() {
     const { buttons, type } = this.props;
     const { activeButtons } = this.state;
     if (type === 'radio') {
       return buttons.map((button, bi) => {
-        const { displayName, value } = button;
+        const { displayName, value } = button; //???
```
Comment (jimexist) [inline src/MDButtonGroup.js]: extra comments

File: src/MDButtonGroup.js [hunk 8]
```diff
@@ -48,6 +62,38 @@ class MDButtonGroup extends Component {
           </button>
         );
       })
+    } else if (type === 'checkbox') {
+      console.log('checkbox');
```
Comment (jimexist) [inline src/MDButtonGroup.js]: no console log

**Comment by jimexist** [comment]
in general there's too many style problems

File: src/MDButtonGroup.js [hunk 9]
```diff
@@ -77,7 +123,7 @@ MDButtonGroup.defaultProps = {
     { displayName: 'button3', value: '3' },
     { displayName: 'button4', value: '4' },
   ],
-  type: 'radio',
+  type: 'checkbox',
 }
```
Comment (ZhengHe-MD) [inline src/MDButtonGroup.js]: 在你测试完以后，应当去掉defaultProps，除非你真的需要设置一些defaultProps。然后把这些props放到App.jsx那里，当做真正的props传递下去

File: src/MDButtonGroup.js [hunk 10]
```diff
@@ -48,6 +62,38 @@ class MDButtonGroup extends Component {
           </button>
         );
       })
+    } else if (type === 'checkbox') {
+      console.log('checkbox');
+      return buttons.map((button, bi) => {
+        const { displayName, value }  = button;
+        let clicked = false;
+        if (activeButtons.has(displayName)){
```
Comment (ZhengHe-MD) [inline src/MDButtonGroup.js]: too much copy&paste here. make className a variable.

File: lib/MDButtonGroup.css (hunk 2 above)
Comment (ZhengHe-MD) [inline lib/MDButtonGroup.css:3]: 这个可以先忽略，我就是让王珂把逻辑先写出来~CSS先忽略，不然内容太多了。

**Pull request merged**